### PR TITLE
PHP 7.4 support

### PR DESCRIPTION
**Pull request by mathieuduffeler**
## Overview

- [x] Added 7.4 support
- [x] Upgraded phpunit
- [x] Added PHPStan
- [x] Added CS-fixer
- [x] Added pre commit hooks
- [x] Setup Github Actions

### CODE REVIEW

File: lib/Sips/Normalizer.php [hunk 1]
```diff
@@ -1,10 +1,11 @@
 <?php
+declare(strict_types=1);
 
 namespace Sips;
 
-class Normalizer
+final class Normalizer
 {
-    private static $invalid = array(
+    private static array $invalid = [
```
Comment (BrechtBonte) [inline lib/Sips/Normalizer.php]: might as well make this a `const`

File: lib/Sips/PaymentRequest.php [hunk 2]
```diff
@@ -56,230 +53,229 @@ class PaymentRequest
...
-	}
-
-    public function setPaymentBrand($brand)
+     * Set amount in cents, eg EUR 12.34 is written as 1234
+     */
+    public function setAmount(int $amount) : void
+    {
+        if (!\is_int($amount)) {
+            throw new InvalidArgumentException("Integer expected. Amount is always in cents");
+        }
```
Comment (ruben-haegeman) [inline lib/Sips/PaymentRequest.php:116]: This can be removed. The type will force it to be an integer

File: lib/Sips/Office/OfficeRequest.php [hunk 3]
```diff
@@ -130,25 +130,24 @@ public function getSignedParameters()
...
     {
-        foreach($this->getRequiredFields($this->getFields()) as $field => $isRequired) {
-            if(
-                !isset($this->parameters[ $field ]) ||
-                $this->parameters[ $field ] === null ||
-                $this->parameters[ $field ] === ''
+        foreach ($this->getRequiredFields($this->getFields()) as $field => $isRequired) {
+            if (!isset($this->parameters[ $field ]) ||
+                $this->parameters[ $field ] === '' ||
+                \is_null($this->parameters[ $field ])
```
Comment (BrechtBonte) [inline lib/Sips/Office/OfficeRequest.php]: `\is_null($this->parameters[ $field ])` is covered by `!isset($this->parameters[ $field ])`

File: lib/Sips/SipsCurrency.php [hunk 4]
```diff
@@ -1,36 +1,42 @@
...
 namespace Sips;
 
 use \InvalidArgumentException;
 
-class SipsCurrency {
+final class SipsCurrency
+{
 
-    private static $currencies = array(
+    private static array $currencies = [
```
Comment (BrechtBonte) [inline lib/Sips/SipsCurrency.php]: const would be good here as well

**Review by BrechtBonte — APPROVED** [review]

File: lib/Sips/Passphrase.php [hunk 5]
```diff
@@ -4,24 +4,18 @@
...
             throw new \InvalidArgumentException("String expected");
         }
         $this->passphrase = $passphrase;
     }
-    
-    /**     
-     * String representation
-     */
+
     public function __toString()
```
Comment (frederikdepaepe) [inline lib/Sips/Passphrase.php]: No return type?

File: lib/Sips/Office/OfficeRequest.php [hunk 6]
```diff
@@ -36,92 +34,91 @@ abstract protected function getMethod();
...
     /** @return string */
-    public function getShaSign()
+    public function getShaSign() : string
     {
         return $this->shaComposer->compose($this->toArray());
     }
 
-    public function setParameter($name, $value)
+    /**
+     * @param Mixed $value
```
Comment (pidera) [inline lib/Sips/Office/OfficeRequest.php]: Parameter type 'mixed' should be lowercase.

```suggestion
     * @param mixed $value
```

File: lib/Sips/ShaComposer/OfficeShaComposer.php [hunk 7]
```diff
@@ -1,41 +1,35 @@
...
 
-    private $ignoredParameters = array('keyVersion', 'seal');
-
-    /**
-     * @param \Sips\Passphrase $passphrase
-     */
     public function __construct(Passphrase $passphrase)
     {
-        $this->passphrase = $passphrase;
+        $this->passphrase = $passphrase->__toString();
```
Comment (pidera) [inline lib/Sips/ShaComposer/OfficeShaComposer.php]: Shouldn't it be better to create an additional method without the underscores if you want to use it explicitly?
```suggestion
        $this->passphrase = $passphrase->toString();
```

**Review by pidera — APPROVED** [review]

**Review by frederikdepaepe — APPROVED** [review]

**Pull request merged**